cmd/skaffold/app/cmd/runner.go

```diff
@@ -39,58 +40,64 @@ import (
 var createRunner = createNewRunner
 
 func withRunner(ctx context.Context, action func(runner.Runner, *latest.SkaffoldConfig) error) error {
-	runner, config, err := createRunner(opts)
+	runner, config, runCtx, err := createRunner(opts)
 	if err != nil {
 		return err
 	}
-
+	event.InitializeState(runCtx.Cfg, runCtx.KubeContext)
+	event.LogMetaEvent()
 	err = action(runner, config)
-
-	return alwaysSucceedWhenCancelled(ctx, err)
+	if err := alwaysSucceedWhenCancelled(ctx, err); err == nil {
+		event.SessionEnded(0)
+		return nil
+	}
+	event.SessionEnded(1)
+	return err
```

Comment:
It seems slightly odd to have the failure case at the end (totally optional)

Reply:
wil flip!